src/iframe/plasmo-storage/plasmo-storage.mock.ts

```diff
@@ -0,0 +1,177 @@
+import { Storage as PlasmoStorage } from "@plasmohq/storage";
+
+export class StorageMock extends PlasmoStorage {
+  constructor() {
+    super({ area: "session" });
+
+    // This browser doesn't support the Storage Access API
+    // so let's just hope we have access!
+    if (!document.hasStorageAccess) return;
+
+    // TODO: Can this be postponed until authentication to avoid requesting permissions too soon?
+    // this.requestStorageAccess();
+
+    // TODO: We need to abstract localStorage as well to replace the hardcoded `localStorage` calls with the handler ones.
+
+    /*
+    TODO: We need a user interaction inside the iframe.
+
+    See https://developer.mozilla.org/en-US/docs/Web/API/Storage_Access_API/Using, https://github.com/GoogleChrome/related-website-sets
+
+    Note: requestStorageAccess() requests are automatically denied unless the embedded content is currently processing
+    a user gesture such as a tap or click (transient activation), or if permission was already granted previously. If
+    permission was not previously granted, requestStorageAccess() requests must be run inside a user gesture-based event
+    handler, as shown above.
+    */
+  }
+
+  /*
+  async requestStorageAccess() {
```

Comment:
This will also be included the library, I've left it here for reference or to add it later if the SDK is ready before the new storage library and we want/need to start using unpartitioned state.